tap_dashboard_reports/streams.py

```diff
@@ -49,19 +54,20 @@ def schema(self) -> dict:
         """Dynamically detect the json schema for the stream.
         This is evaluated prior to any records being retrieved.
         """
-
         properties = [
             th.Property(field, th.StringType)
             for field in (self._dimensions + self._measures)
         ]
-
+        if self.custom_key:
+            properties.append(th.Property(self.primary_keys, th.StringType))
         # Return the list as a JSON Schema dictionary object
         return th.PropertiesList(*properties).to_dict()
 
     def get_records(self, context):
         columns = self._dimensions + self._measures
+        if len(self.primary_keys) != 0:
```

Comment:
This logic is incorrect as `self.primary_keys` contains `self._dimensions`, and `self._dimensions` already added to columns in the row above

Reply:
commit custom_key check up for this line